File: Paid_ChatGPT/sistemaDeTiposRevisado/tipos.js
function convertirANumero(valor) {
    if (typeof valor === "number") {
        return valor;
    }

    if (typeof valor === "string") {
        return Number(valor);
    }

    throw new Error("Tipo no soportado");
}

function promedioMayoresAUmbral(arreglo, umbral) {
    const umbralNum = convertirANumero(umbral);

    let suma = 0;
    let cantidad = 0;

    for (const elemento of arreglo) {
        const valor = convertirANumero(elemento);

        if (valor > umbralNum) {
            suma += valor;
            cantidad++;
        }
    }

    return cantidad === 0 ? 0 : suma / cantidad;
}

// Ejemplo de uso
const valores = [1, 2.5, 30, "40", 50];

console.log(promedioMayoresAUmbral(valores, 25));

File: Paid_ChatGPT/sistemaDeTiposRevisado/tipos.py
def promedio_mayores_a_umbral(arreglo, umbral):
    suma = 0.0
    cantidad = 0

    umbral_num = float(umbral)

    for elemento in arreglo:
        valor = float(elemento)

        if valor > umbral_num:
            suma += valor
            cantidad += 1

    if cantidad == 0:
        return 0

    return suma / cantidad